### PR TITLE
[RFR] Upgrade to material-ui 1.2.1

### PR DESCRIPTION
**Pull request by fzaninotto**
Normally no breaking changes, but it adds new features. See https://github.com/mui-org/material-ui/releases for detailed changelog.

### CODE REVIEW

File: examples/demo/src/products/GridList.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ const styles = {
             'linear-gradient(to top, rgba(0,0,0,0.8) 0%,rgba(0,0,0,0.4) 70%,rgba(0,0,0,0) 100%)',
     },
     link: {
-        color: 'white',
+        color: '#fff!important',
```
Comment (djhi) [inline examples/demo/src/products/GridList.js]: :scream:

Comment (fzaninotto) [reply examples/demo/src/products/GridList.js]: yeah, I know. If you find a better way, I'm your man.

Comment (djhi) [reply examples/demo/src/products/GridList.js]: Well, it works without `!important` on chrome

Comment (fzaninotto) [reply examples/demo/src/products/GridList.js]: not on my laptop with the latest material ui

Comment (djhi) [reply examples/demo/src/products/GridList.js]: I used this branch and did not forget to `make install`

**Pull request merged**